department-of-veterans-affairs/caseflow
# [PART 2] CaseTitleDetails changes & FeatureToggle

Bumps #13914

PART 2 of stack
PART 1 #14331
PART 3 #14332

### Description
Integrate the Mark/Remove overtime button into `CaseTitleDetails`.  Changes include:
- create selectors to check for legacy tasks associated with user
- bring in the `:overtime_revamp` FeatureToggle
- add the Mark/Remove case as overtime button
- create clock SVG icon for use alongside the button
- add logic to check user roles and appeal status to conditionally render the button
- add the route which triggers the modal from button interaction

### Acceptance Criteria
- [ ] Work is behind the feature toggle: :overtime_revamp
- [ ] The feature should be accessible to the following user groups: VLJs (SSCs are VLJs in Caseflow)

### Testing Plan

1. enable `overtime_revamp` feature toggle 
```ruby
FeatureToggle.enable!(:overtime_revamp)
```
2. Go to http://localhost:3000/queue as BVAAABSHIRE
3. Select a legacy appeal
4. Ensure you can see the option to Mark this case as overtime
5. Repeat steps 2 & 3 for an ama appeal
5. Go the assign view and assign the legacy and ama cases you viewed to an attorney on BVAAABSHIRE's team
6. Login as that attorney
7. Ensure you cannot see the option to Mark this case as overtime on either appeal

### Legacy assigned to judge

### Before
<img width="1438" alt="Screen Shot 2020-05-14 at 10 19 32 AM" src="https://user-images.githubusercontent.com/18618189/81945714-72cb1400-95cc-11ea-8d1a-ca350cf11251.png">

### After
<img width="1440" alt="Screen Shot 2020-05-20 at 10 55 29 AM" src="https://user-images.githubusercontent.com/18618189/82461526-88d34b80-9a88-11ea-8399-c93eb2732dad.png">


### Ama assigned to judge

### Before
<img width="1440" alt="Screen Shot 2020-05-20 at 11 01 28 AM" src="https://user-images.githubusercontent.com/18618189/82462239-56761e00-9a89-11ea-889a-253410aa891e.png">

### After
<img width="1439" alt="Screen Shot 2020-05-20 at 11 01 37 AM" src="https://user-images.githubusercontent.com/18618189/82462256-5a09a500-9a89-11ea-9bb0-7c01194cab80.png">


### Legacy assigned to attorney

### Before
<img width="1439" alt="Screen Shot 2020-05-20 at 11 28 21 AM" src="https://user-images.githubusercontent.com/18618189/82465304-203a9d80-9a8d-11ea-93e1-f9e6976d95d1.png">

### After
<img width="1440" alt="Screen Shot 2020-05-20 at 11 10 02 AM" src="https://user-images.githubusercontent.com/18618189/82465336-2892d880-9a8d-11ea-99c5-f9c5c1610c44.png">

### Ama assigned to attorney

### Before
<img width="1440" alt="Screen Shot 2020-05-20 at 11 28 31 AM" src="https://user-images.githubusercontent.com/18618189/82465377-35afc780-9a8d-11ea-959c-65b90fa5849f.png">

### After
<img width="1440" alt="Screen Shot 2020-05-20 at 11 10 11 AM" src="https://user-images.githubusercontent.com/18618189/82465419-3e080280-9a8d-11ea-8964-79201070279e.png">
